src/app/service/general.ts

```diff
@@ -18,6 +18,11 @@ export class GeneralService {
     return this.httpClient.get<Coupon[]>(url);
   }
 
+  getCoupon(id: number): Observable<Coupon> {
+    const url = `http://localhost:8080/general/coupon/${id}`;
+    return this.httpClient.get<Coupon>(url);
+  }
+
```

Comment:
regarding this file, i would call it "adding/added getCoupon function to GeneralService"

Reply:
okay, but comment on my code and not my commit names lol